From 0726a373d5a4416fbda15bd1c03ccfe96e41b342 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 19 Jul 2016 17:36:03 +0300
Subject: [PATCH 0616/2241] fix: mmc: enable clock gatting in eMMC/SD driver

Enable clock gatting in eMMC/SD driver to disable clock when
there's no data transfer on SDIO/eMMC interface to save power

Change-Id: I9514005339def2cbbda522f52d18b91eb0ec0abc
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31249
Reviewed-by: Victor Gu <xigu@marvell.com>
Verified-Armada37x0: Victor Gu <xigu@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 drivers/mmc/host/sdhci-xenon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/sdhci-xenon.c b/drivers/mmc/host/sdhci-xenon.c
index 8048566..888e38a 100644
--- a/drivers/mmc/host/sdhci-xenon.c
+++ b/drivers/mmc/host/sdhci-xenon.c
@@ -893,9 +893,9 @@ static void sdhci_xenon_reset_exit(struct sdhci_host *host,
 	/* Disable tuning request and auto-retuing again */
 	sdhci_xenon_setup_tuning(host);
 
-	sdhci_xenon_set_acg(host, false);
+	sdhci_xenon_set_acg(host, true);
 
-	sdhci_xenon_set_sdclk_off_idle(host, slotno, false);
+	sdhci_xenon_set_sdclk_off_idle(host, slotno, true);
 }
 
 static void sdhci_xenon_reset(struct sdhci_host *host, u8 mask)
-- 
2.7.4

